student-os/scripts/probe_materials.py

```
@@ -495,8 +496,33 @@ def resolve_probe(
+    # upload and split as needed for PDFs.
+    if probe.get("needs_api") and not has_api_token and method == "api" and not force_strategy and suffix in API_SUPPORTED_SUFFIXES:
+        probe = {
+            **probe,
+            "strategy": "mineru-agent-v1",
+            "tool": "mineru-api",
+            "needs_api": False,
+            "agent_api": True,
+            "agent_api_limit_bytes": MINERU_AGENT_MAX_FILE_BYTES,
+            "reason": f"{probe.get('reason')}; MinerU token unavailable, using MinerU v1 Agent API",
+        }
+        return probe
```

Ruff (0.16.1)

[warning] Unnecessary assignment to `probe` before `return` statement

Remove unnecessary assignment

(RET504)